R/functions.R

```diff
@@ -1004,22 +1015,51 @@ get_co2_price_global_tmp = function() {
 #' Get co2 price between CO2 and CO2_ETS. If only one CO2 type present, share = 1;
 #' otherwise, each type has the share corresponding to the last historical year
 #' @keywords co2 rgcam::getQuery tmp
-#' @return co2_price_share global variable
+#' @return co2_price_share_byreg and co2_price_share_bysec global variables
 #' @export
 get_co2_price_share = function() {
-  co2_price_share <<- co2_clean %>%
+  co2_price_share_byreg <<- co2_clean %>%
     dplyr::filter(var == 'Emissions|CO2|Energy and Industrial Processes',
                   year == last_historical_year) %>%
-    rbind(co2_ets %>%
+    rbind(co2_ets_byreg %>%
             dplyr::filter(var == 'Emissions|CO2_ETS|Energy and Industrial Processes',
                           year == last_historical_year)) %>%
-    dplyr::group_by(scenario, region, year) %>%
-    dplyr::mutate(share = value / sum(value)) %>%
-    dplyr::rename(market_adj = var) %>%
-    dplyr::mutate(market_adj = dplyr::if_else(market_adj == "Emissions|CO2|Energy and Industrial Processes", 'CO2', 'CO2_ETS')) %>%
-    dplyr::ungroup() %>%
-    # if CO2, share = 1
-    dplyr::mutate(share = dplyr::if_else(market_adj == 'CO2', 1, share))
+    dplyr::mutate(var = dplyr::if_else(var == "Emissions|CO2|Energy and Industrial Processes", 'CO2', 'CO2_ETS')) %>%
```

Comment:
Revisar esto, porque yo entendí que si existe el ETS se aplica a "Energy and Industrial Processes" y a "Supply". Entendí que con el nuevo mapping esto se aplica automático, por eso no entiendo muy bien que hace esto... (aunque estará bien! sólo por confirmar)

Reply:
"energy and industrial processes" es todo el CO2 fossil, era el variable que se usa para weight.